browser/brave_content_browser_client.cc

```diff
@@ -347,3 +333,42 @@ GURL BraveContentBrowserClient::GetEffectiveURL(
   return url;
 #endif
 }
+
+// [static]
+bool BraveContentBrowserClient::HandleURLOverrideRewrite(GURL* url,
+    content::BrowserContext* browser_context) {
+  // redirect sync-internals
+  if (url->host() == chrome::kChromeUISyncInternalsHost ||
+      url->host() == chrome::kChromeUISyncHost) {
+    GURL::Replacements replacements;
+    replacements.SetHostStr(chrome::kChromeUISyncHost);
+    *url = url->ReplaceComponents(replacements);
+    return true;
+  }
+
+  // no special win10 welcome page
+  if (url->host() == chrome::kChromeUIWelcomeWin10Host ||
+      url->host() == chrome::kChromeUIWelcomeHost) {
+    *url = GURL(chrome::kChromeUIWelcomeURL);
+    return true;
+  }
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+  if (url->SchemeIs(content::kChromeUIScheme) &&
+      url->host() == ethereum_remote_client_host) {
+    if (browser_context) {
```

Comment:
as mentioned in dm there is a race condition here with loading the url vs loading the extension because they happen on different threads

Reply:
didn't realize the extension was already installed at this point so nothing async

Reply:
Changed this to use a throttle now, avoids possible race condition from renderer for sure that way.